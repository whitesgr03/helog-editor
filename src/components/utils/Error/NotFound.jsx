// Styles
import style from "../../styles/layout/Error.module.css";
import image from "../../styles/utils/image.module.css";

const NotFound = () => {
	return (
		<div className={style.error}>
			<span className={`${image.icon} ${style.alert}`} />
			<h2>Page Not Found</h2>
			<div className={style.message}>
				<p>Our apologies, there has been an error.</p>
				<p>The page you are looking for could not be found.</p>
				<p>You may surf over to our other pages.</p>
			</div>
		</div>
	);
};

export default NotFound;
