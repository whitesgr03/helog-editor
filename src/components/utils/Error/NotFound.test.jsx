import { expect, describe, it } from 'vitest';
import { render } from '@testing-library/react';

import { RouterProvider, createMemoryRouter } from 'react-router-dom';

import { NotFound } from './NotFound';

describe('NotFound component', () => {
	it('should match snapshot', () => {
		const router = createMemoryRouter(
			[
				{
					path: '/',
					element: <NotFound />,
				},
			],
			{
				future: {
					v7_relativeSplatPath: true,
				},
			},
		);

		const { asFragment } = render(
			<RouterProvider
				router={router}
				future={{
					v7_startTransition: true,
				}}
			/>,
		);

		const actual = asFragment();

		expect(actual).toMatchSnapshot();
	});
});
