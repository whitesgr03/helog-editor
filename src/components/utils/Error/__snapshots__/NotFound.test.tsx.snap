// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="_error_48ab16"
  >
    <span
      class="_icon_620964 _alert_48ab16"
    />
    <h2>
      Page Not Found
    </h2>
    <div
      class="_message_48ab16"
    >
      <p>
        Our apologies, there has been an error.
      </p>
      <p>
        The page you are looking for could not be found.
      </p>
      <p>
        You may surf over to our other pages.
      </p>
    </div>
    <a
      class="_link_48ab16"
      href="/"
    >
      Back to Dashboard Page
    </a>
  </div>
</DocumentFragment>
`;
